adapter/internal/discovery/xds/server.go

```diff
@@ -171,60 +167,91 @@ func GetEnforcerApplicationKeyMappingCache() wso2_cache.SnapshotCache {
+	//TODO: Uncomment the below section depending on MgwSwagger.Resource ids
+	//existingMgwSwagger, exists := apiMgwSwaggerMap[apiIdentifier]
+	// if exists {
+	// 	if reflect.DeepEqual(mgwSwagger, existingMgwSwagger) {
+	// 		logger.LoggerXds.Infof("API %v already exists. No changes to apply.", apiIdentifier)
+	// 		return
+	// 	}
+	// }
+	apiMgwSwaggerMap[apiIdentifier] = mgwSwagger
 	//TODO: (VirajSalaka) Handle OpenAPIs which does not have label (Current Impl , it will be labelled as default)
 	// TODO: commented the following line as the implementation is not supported yet.
 	//newLabels = model.GetXWso2Label(openAPIV3Struct.ExtensionProps)
 	//:TODO: since currently labels are not taking from x-wso2-label, I have made it to be taken from the method
 	// argument.
-	newLabels = environments
-	logger.LoggerXds.Infof("Added/Updated the content under OpenAPI Key : %v", apiMapKey)
-	logger.LoggerXds.Debugf("Newly added labels for the OpenAPI Key : %v are %v", apiMapKey, newLabels)
-	oldLabels, _ := openAPIEnvoyMap[apiMapKey]
-	logger.LoggerXds.Debugf("Already existing labels for the OpenAPI Key : %v are %v", apiMapKey, oldLabels)
-	openAPIEnvoyMap[apiMapKey] = newLabels
-
-	routes, clusters, endpoints := oasParser.GetProductionRoutesClustersEndpoints(mgwSwagger, upstreamCerts)
+	newLabels = apiContent.Environments
+	logger.LoggerXds.Infof("Added/Updated the content under OpenAPI Key : %v", apiIdentifier)
+	logger.LoggerXds.Debugf("Newly added labels for the OpenAPI Key : %v are %v", apiIdentifier, newLabels)
+	oldLabels, _ := openAPIEnvoyMap[apiIdentifier]
+	logger.LoggerXds.Debugf("Already existing labels for the OpenAPI Key : %v are %v", apiIdentifier, oldLabels)
+	openAPIEnvoyMap[apiIdentifier] = newLabels
+
+	routes, clusters, endpoints := oasParser.GetProductionRoutesClustersEndpoints(mgwSwagger, apiContent.UpstreamCerts)
 	// TODO: (VirajSalaka) Decide if the routes and listeners need their own map since it is not going to be changed based on API at the moment.
-	openAPIRoutesMap[apiMapKey] = routes
+	openAPIRoutesMap[apiIdentifier] = routes
 	// openAPIListenersMap[apiMapKey] = listeners
-	openAPIClustersMap[apiMapKey] = clusters
-	openAPIEndpointsMap[apiMapKey] = endpoints
-	openAPIEnforcerApisMap[apiMapKey] = oasParser.GetEnforcerAPI(mgwSwagger)
+	openAPIClustersMap[apiIdentifier] = clusters
+	openAPIEndpointsMap[apiIdentifier] = endpoints
+	openAPIEnforcerApisMap[apiIdentifier] = oasParser.GetEnforcerAPI(mgwSwagger)
 	// TODO: (VirajSalaka) Fault tolerance mechanism implementation
 	updateXdsCacheOnAPIAdd(oldLabels, newLabels)
 	if svcdiscovery.IsServiceDiscoveryEnabled {
 		startConsulServiceDiscovery() //consul service discovery starting point
 	}
 }
 
+// DeleteAPI deletes an API, its resources and updates the caches
+func DeleteAPI(apiName string, version string, vhost string) error {
+	apiIdentifier := vhost + ":" + apiName + ":" + version
```

Comment:
Shall we introduce a new method to generation of apidentifier as It is used twice

Reply:
Introducing a new method to generate the apidentifier